src/components/Badge/Badge.jsx

```diff
@@ -94,8 +94,12 @@ const Badge = props => {
                         props.userProfile.badgeCollection.find(
                           badgeObj => badgeObj.badge.type === 'Personal Max',
                         )
-                          ? ` and a personal best of ${props.userProfile.personalBestMaxHrs} ${
-                              props.userProfile.personalBestMaxHrs === 1 ? 'hour' : 'hours'
+                          ? ` and a personal best of ${parseFloat(
```

Comment:
In my opinion, would be great if you created a function and put all logic that is related to `totalBadge` there. I would divide UI and JS logic. Although this is my opinion

Reply:
Thanks for your valuable suggestion! I just pushed a commit that extracts the badge text logic to a separate function. I tested locally and the text in the badge section works. Could you please also test to see if it works for you?